### PR TITLE
VMs List View

### LINKED ISSUE #224: User should be able to drill into the Migration Plan details from the Migrations Not Started List View
Allowing the user to click anywhere on a list view item in the **Migrations Not Started** List View to drill into the migration plan details.

@vconzola FYI

For implementation reference see: https://github.com/priley86/miq_v2v_ui_plugin/issues/218

### PR DESCRIPTION
**Pull request by michaelkro**
* Clicking on a "Migration Plan Not Started" transitions to `/migration/plan/:plan_id`, where a list of VMs for the plan is displayed
* When entering `/migration/plan/:plan_id`
  1. We first fetch the `transformation plan` with its `miq_requests`
  2. If there are any `miq_requests`, we know that plan has been started, so we fetch the most recent request to display its `miq_request_tasks` and start polling
  3. If there are no requests, we know that the plan is still in the "queued" state, so we query the API for the plan's VMs using the ids from `config_info`, and display the list
* In order to construct API calls with dynamic segments, update `/javascript/components/index.js`
* Reuse `<PlanRequestDetailsList>`, modified, to display the VMs
* Closes #224 

## Notes
1. Our `<Sort>` component does not allow there to be only one sort criterion, so in the screenshot below, notice there is no button next to the `asc/desc` toggle
2. To my knowledge, we do not have an API that can query a plan for both its VM names and storages info, so for now I just put `0 GBs migrated` as a placeholder (as compared to the mock provided in https://github.com/priley86/miq_v2v_ui_plugin/issues/224)
## Screens
fullscreen_4_23_18__4_11_pm

### CODE REVIEW

**Comment by priley86** [comment]
this looks mostly good as far as the code goes to me (and the logic seems correct).

a couple questions:
* @AparnaKarve can you confirm that we should use the `/api/vms` API to query the vms for the plans which have no requests?
* I noticed the same issue regarding `Sort` (it won't display a dropdown if there is only one item to sort on). I'm wondering if we should change the component in PF React to accept one sort indicator. Otherwise it seems kind of confusing what I am sorting on.

File: app/javascript/react/screens/App/Plan/Plan.js [hunk 1]
```diff
@@ -5,55 +5,78 @@ import { Link } from 'react-router-dom';
...
-  //   }
-  //   return {
-  //     planRequestTasks,
-  //     planRequestTasksMutable: Immutable.asMutable(planRequestTasks)
-  //   };
-  // }
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { planRequestTasks, vms } = nextProps;
+    const { planInProgress } = prevState;
+    if (planInProgress && planRequestTasks === prevState.planRequestTasks) {
```
[the remark this answers is not in the document]
Comment (michaelkro) [reply app/javascript/react/screens/App/Plan/Plan.js]: Ok, yes, I agree with you.  Will make the change :)

**Comment by vconzola** [comment]
@priley86  Regarding sort, how are the VMs names populated in the table by default? If they are already alphabetical, we may not even need a sort?

**Comment by AparnaKarve** [comment]
@priley86 The API to query VMs in general is - `/api/vms`, and  has been used here correctly in the bulk query POST call.

@michaelkro This looks good!

**Comment by priley86** [comment]
ok this appears to be correct after discussing w/ @jeff-phillips-18 . We should not show a dropdown if there is only a single sort category. So i'm good with this... approving.

**Pull request merged**